include/openenclave/tracee.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_TRACEE_H
+#define _OE_TRACEE_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/enclave.h>
+#include <openenclave/log.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Define callback to process logs generated by enclave.
+ *
+ * The callback, defined as expected, will be called every time OE_TRACE is
+ * is called by enclave routines, irrespective of log level. The callback will
+ * be processed under log lock.
+ *
+ * NOTE: The function oe_enclave_log_set_callback sets the callback and only
+ * then will enclave logs be processed through the callback function.
+ *
+ * @param[in] context The parameter can be used by the user to send in an
+ * additional variable or structures that might be needed to process the logs.
+ * @param[in] level The parameter holds the log level of the current enclave
+ * log being processed and can be used by the callback.
+ * @param[in] thread_id The parameter holds the thread id of the curent
+ * processing thread.
+ * @param[in] message The parameter holds the enclave log message generated.
+ */
+typedef void (*oe_enclave_log_callback_t)(
+    void* context,
+    oe_log_level_t level,
+    uint64_t thread_id,
+    const char* message);
+
+/**
+ * Set callback that will process logs generated by the enclave.
+ *
+ * This function, when called ensures the enclave logs are transferred to
+ * the preferred callback instead of the regular route.
+ *
+ * @param[in] context A pointer to variable or structure that can be used to
+ * process the logs inside the callback function.
+ * @param[in] callback The parameter holds pointer to function that is be
+ * assigned as the callback to process enclave logs.
+ *
+ * @retval OE_OK Callback was added successfully.
+ * @retval OE_FAILURE Failed to add callback.
+ */
+oe_result_t oe_enclave_log_set_callback(
+    void* context,
```

Comment:
From the API perspective, I would suggest add the `context_size` variable so that the function don't blindly take an opaque pointer.

Reply:
I like this suggestion. Would be safer to add size of the structure, if any.

@achamayou, do you think its okay to deviate from host callback API and have this additional parameter in the set_callback API?

Reply:
@RRathna sure, that's fine. Ultimately the implementation needs to know what to cast to, but having the size does provide some safety.

Reply:
Advice from me as the API maintainer: If the context is always being passed back to the same code that set the context then you do not need a context size.  If the caller cares about the size, it can encode it in the first field of the context structure.   Unless more than one piece of code (i.e., from different developers) needs it, one should not add a context size.

Reply:
I'm fine with Dave's suggestion. I guess it's just unclear to me how the context could be used. It would be helpful to demonstrate some use cases in the sample as well (instead of passing NULL in the current sample).

Reply:
In cases where the user app wants to log to something other than a file, this may point to a network context for example, or a message queue.

Reply:
Example: app calls the set callback function and passes in as context, a pointer to a global struct.   In the callback, app casts the context pointer back to its own struct pointer.

Reply:
Will add sample with context set to a global structure in a seperate PR.